refaster-runner/src/test/java/tech/picnic/errorprone/refaster/runner/RefasterTest.java

```diff
@@ -0,0 +1,243 @@
+package tech.picnic.errorprone.refaster.runner;
+
+import static com.google.common.base.Predicates.containsPattern;
+import static com.google.common.collect.ImmutableSortedMap.toImmutableSortedMap;
+import static java.util.Comparator.naturalOrder;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.junit.jupiter.params.provider.Arguments.arguments;
+
+import com.google.common.collect.ImmutableList;
+import com.google.errorprone.BugCheckerRefactoringTestHelper;
+import com.google.errorprone.BugCheckerRefactoringTestHelper.TestMode;
+import com.google.errorprone.CompilationTestHelper;
+import java.lang.reflect.Modifier;
+import java.util.Arrays;
+import java.util.regex.Pattern;
+import java.util.stream.Stream;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+final class RefasterTest {
+  private final CompilationTestHelper compilationHelper =
+      CompilationTestHelper.newInstance(Refaster.class, getClass())
+          .matchAllDiagnostics()
+          .expectErrorMessage(
+              "StringOfSizeZeroTemplate",
+              containsPattern(
+                  "\\[Refaster Rule\\] FooTemplates\\.StringOfSizeZeroTemplate: Refactoring opportunity\\s+.+\\s+null"))
```

Comment:
According to the `Description.Builder`, `link` is `Nullable`. We could instead of an empty string pass in `null` at `AnnotatedCompositeCodeTransformer#augmentDescription` if we don't have a `url`, causing this to not appear at all (verified this). This might be nice to do in this transition period until all checks and templates have a link. Got a commit ready if we wish to do this.

Reply:
> Wanna open a PR upstream? :)

Yes that sounds good. Would be a nice improvement.
Will write it down and do after the EPS OSS-ing journey 😛.